src/WebJobs.Script.WebHost/HostNameProvider.cs

```diff
@@ -52,17 +50,17 @@ public virtual string Value
             }
         }
 
-        public virtual void Synchronize(HttpRequest request)
+        public virtual void Synchronize(HttpRequest request, ILogger logger)
```

Comment:
Should this be enforced as type` ILogger<HostnameFixupMiddleware>` ?

Reply:
I think because this is a public method in `HostNameProvider`, which can be called by anyone, we may not want to enforce the current caller's type. If this was in `HostNameFixupMiddleware` I would agree.